### PR TITLE
More work, more models

### PR DESCRIPTION
**Pull request by dotemacs**
I've created all but two models for the whole app. Why? So that we can see where the repetition is (because there will be some, and there is) and to get this app done quicker.

Now what's left:

* adding the controllers
* adding the views
* sorting out the persistence (into the session)

The missing models are:

* Contact: http://ministryofjustice.github.io/fr-prototypes/public/journey-7/contact.html
* DOB: http://ministryofjustice.github.io/fr-prototypes/public/journey-7/dob.html

### CODE REVIEW

File: app/models/probate_case_fee.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class ProbateCaseFee < Base
+  attribute :answer, Boolean
+
+  validates :answer, inclusion: { in: [true, false] }
+end
```
Comment (colinbruce) [inline app/models/probate_case_fee.rb]: Doesn't this need name and dob of deceased too?

[Link](http://ministryofjustice.github.io/fr-prototypes/public/journey-7/probate.html)

File: app/models/fee_paid.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class FeePaid < Base
+  attribute :answer, Boolean
+
+  validates :answer, inclusion: { in: [true, false] }
```
Comment (colinbruce) [inline app/models/fee_paid.rb]: This should have a date fee paid too

[link](http://ministryofjustice.github.io/fr-prototypes/public/journey-7/refund.html)

File: app/models/probate_case_fee.rb (hunk 1 above)
Comment (dotemacs) [inline app/models/probate_case_fee.rb]: Ah, didn't spot it

File: app/models/fee_paid.rb (hunk 2 above)
Comment (dotemacs) [inline app/models/fee_paid.rb]: Good catch.

File: app/models/case_number.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class CaseNumber < Base
+  attribute :number, Boolean
+
+  validates :number, inclusion: { in: [true, false] }
+end
```
Comment (colinbruce) [inline app/models/case_number.rb]: Needs a case number field

[Link](http://ministryofjustice.github.io/fr-prototypes/public/journey-7/case-number.html)

File: app/controllers/home_controller.rb [hunk 4]
```diff
@@ -1,17 +1,25 @@
...
+    end
 
-  def marital_status_save
-    session[:marital_status] = params["marital_status"]["married"]
-    redirect_to :summary
+    define_method("#{action}_save") do
+      session[action] = params[action.to_s]["married"]
+      redirect_to :summary
+    end
   end
```
Comment (dotemacs) [inline app/controllers/home_controller.rb]: What is meant to happen here:

* create a **show** action
* create a **save** action

Both from the name of the model, which is stored in **ACTION**.

**Comment by colinbruce** [comment]
Looks fine as per discussion.

**Pull request merged**